apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts

```diff
@@ -108,13 +115,32 @@ export class VatLiabilitiesPayments implements OnInit, OnDestroy {
             this.isPaymentMode = this.router.routerState.snapshot.url.includes('payments');
             this.displayColumns = this.isPaymentMode ? this.paymentColumns : this.liabilityColumns;
             this.noDataFound = true;
-            if (this.getFormControl('taxNumber').value) {
+        });
+        this.loadTaxDetails();
+        this.taxNumber$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.body?.length) {
+                this.taxesList = response.body.map(tax => ({
+                    label: tax,
+                    value: tax
+                }));
+                if (this.taxesList.length === 1) {
+                    this.getFormControl('taxNumber').patchValue(this.taxesList[0].value);
+                }
                 this.getLiabilitiesPayment();
             }
-        })
-        this.loadTaxDetails();
-        this.isCompanyMode = this.generalService.currentOrganizationType === OrganizationType.Company;
+        });
+        this.liabilityPaymentList$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (this.isPaymentMode && response?.body?.payments || (!this.isPaymentMode) && response?.body?.liabilities) {
```

Comment:
if ((this.isPaymentMode && response?.body?.payments) || ((!this.isPaymentMode) && response.body?.liabilities)) {